code/datums/components/bumpattack.dm

```diff
@@ -0,0 +1,72 @@
+/**
+ * Add an auto attack on bump behaviour to item
+ * valid_slot: the inventory slot the item could be held in while still bumpattacking with it
+ * proxy_weapon: the weapon that will gain this behaviour
+ */
+#define COOLDOWN_BUMP_ATTACK "bump_attack_cooldown"
+
+/datum/component/bumpattack
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///inventory slot that the item could be stored while still being able to attack with it
+	var/list/valid_slots
+	///whether or not the behaviour would be active on the item
+	var/active = FALSE
+	///the mob handling the item
+	var/mob/living/wearer
+	///the item that will gain this behaviour
+	var/obj/item/proxy_weapon
+	///cool down between each hit
+	var/attack_cooldown = CLICK_CD_MELEE
+
+/datum/component/bumpattack/Initialize(valid_slots, obj/item/proxy_weapon)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.valid_slots = valid_slots
+	src.proxy_weapon = proxy_weapon
+
+	return ..()
+
+/datum/component/bumpattack/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/check_equip)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/check_drop)
+
+/datum/component/bumpattack/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED))
+
+/datum/component/bumpattack/proc/check_equip(obj/item/source, mob/living/user, slot)
+	SIGNAL_HANDLER
+	if(!user) // iunno, thoroughness
+		return
+	if(slot & valid_slots)
```

Comment:
You can't check the slot of a list

Wrong:
```
(slot & list(slot1. slot2))
```

Functional:
```
(slot & (slot1|slot2))
```

You'll have make this not a list for it to properly work

Reply:
i just made it a var and update the comment to not use a list for multiple inventory slot